src/objects/creep_shared.rs

```diff
@@ -262,14 +262,79 @@ pub unsafe trait SharedCreepProperties: RoomObjectProperties {
 unsafe impl SharedCreepProperties for Creep {}
 unsafe impl SharedCreepProperties for PowerCreep {}
 
+#[derive(Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub enum LineStyle {
+    Undefined,
```

Comment:
I don't think this will serialize to the right value - it will make the string "undefined", rather than an actual JavaScript undefined value.

The only way I know of to actually serialize to `undefined`, though, is by leaving a value out entirely. I think we could do this if we added a `is_undefined()` method to `LineStyle`, and used [`skip_serializing_if`](https://serde.rs/field-attrs.html#skip_serializing_if) like so in `PolyStyle`:

```rust
#[serde(skip_serializing_if = "LineStyle::is_undefined")]
line_style: LineStyle
```